language/testing-infra/e2e-tests/src/common_transactions.rs

```diff
@@ -72,6 +72,95 @@ pub static EMPTY_SCRIPT: Lazy<Vec<u8>> = Lazy::new(|| {
         .expect("Failed to compile")
 });
 
+pub static MULTI_AGENT_SWAP_SCRIPT: Lazy<Vec<u8>> = Lazy::new(|| {
+    let code = "
+    import 0x1.DiemAccount;
+    import 0x1.Signer;
+    import 0x1.XDX;
+    import 0x1.XUS;
+
+    // Alice and Bob agree on the value of amount_xus and amount_xdx off-chain.
+    main(alice: signer, bob: signer, amount_xus: u64, amount_xdx: u64) {
```

Comment:
Not sure if you want to do this in this PR, or if you want to do it in a followup but I think now that we have non-`&signer` args to scripts, having multi-agent scripts have a signature like `main(primary_signer: signer, secondary_signers: vector<signer>, ...)` is probably what we will want at the end.

Reply:
I think it would make to add some tests that `public(script) fun f(accounts: vector<signer>)` works correctly in this PR, though.

Reply:
Makes sense. I'll add a test that uses vector<signer>.

Reply:
I talked to @tnowacki and it seems that we don't currently support executing functions that take in `vector<signer>`. https://github.com/diem/diem/blob/main/language/move-vm/runtime/src/runtime.rs#L179

Reply:
Gotcha. Then it probably doesn't make sense to add it in this PR, but I wonder if it's something we should consider supporting in the near future. Some interesting use-cases (e.g., K-of-N approvals for an on-chain action) would require this. Thoughts @tnowacki?

Reply:
Yes. @tzakian and I both thought supporting this could be a nice follow up task after this PR lands.